### DISCUSSION TITLE
Elsa 2.9.0

### DISCUSSION THREAD

**Discussion by sfmskywalker** [elsa-workflows/elsa-core#3360] [Design & tech]
## Known Issues

* Deleting entities throws an Entity Framework exception. This will be fixed in 2.9.1 and a fix is already available on MyGet in the preview packages.

## What's Changed
* Liquid is liquid by @mohdali in https://github.com/elsa-workflows/elsa-core/pull/3107
* Add Persian language localization by @VahidNaderi in https://github.com/elsa-workflows/elsa-core/pull/3117
* Remove triggers that have a malformed path by @AdamJohnSwan in https://github.com/elsa-workflows/elsa-core/pull/3121
* Add XML Code Comments For The HTTP Activities by @AdamJohnSwan in https://github.com/elsa-workflows/elsa-core/pull/3127
* Workflow Definitions Search by @andreabalducci in https://github.com/elsa-workflows/elsa-core/pull/3131
* Inject the Authentication handler to override the anonymous handler t… by @r-ulak in https://github.com/elsa-workflows/elsa-core/pull/3139
* Persist blob metadata changes by @stuartmcgillivray in https://github.com/elsa-workflows/elsa-core/pull/3116
* Define inputs in reusable objects by @stuartmcgillivray in https://github.com/elsa-workflows/elsa-core/pull/3148
* Generic cascading drop down by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/3063
* Mongo db can't write to read only properties by @stuartmcgillivray in https://github.com/elsa-workflows/elsa-core/pull/3150
* [RabbitMQ] Add Javascript Syntax to Headers definition by @borjaferv in https://github.com/elsa-workflows/elsa-core/pull/3151
* Add "EagerJoin" option to joins by @martinhjulstrom in https://github.com/elsa-workflows/elsa-core/pull/3152
* Check if attachments object is empty string by @johnwc in https://github.com/elsa-workflows/elsa-core/pull/3164
* Mongo serializer registration by @stuartmcgillivray in https://github.com/elsa-workflows/elsa-core/pull/3170
* Publish symbol packages by @holidayfun in https://github.com/elsa-workflows/elsa-core/pull/3181
* If eager join is set, clear scheduled activities on join by @martinhjulstrom in https://github.com/elsa-workflows/elsa-core/pull/3177
* Jint options configuration by @stuartmcgillivray in https://github.com/elsa-workflows/elsa-core/pull/3178
* don't lowercase the Path Parameters in the URI (#3187) by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/3189
* add support spanish language. add i18n translate in menu Items by @Juandavi1 in https://github.com/elsa-workflows/elsa-core/pull/3190
* Fault array, fix retry, new parameter in Join activity by @smartwaystudio in https://github.com/elsa-workflows/elsa-core/pull/3182
* Restore Fault property and mark it as obsolete. Patched it for less impact (ref PR 3182) by @smartwaystudio in https://github.com/elsa-workflows/elsa-core/pull/3194
* Hotfix/docker image base platform by @milkyware in https://github.com/elsa-workflows/elsa-core/pull/3204
* Feature/i18n de by @SebastianWachsmuth in https://github.com/elsa-workflows/elsa-core/pull/3212
* Fix Server Sample Swagger Gen by @mohdali in https://github.com/elsa-workflows/elsa-core/pull/3215
* Adding Statements to Catch if expression has no value.  by @Dioltas95 in https://github.com/elsa-workflows/elsa-core/pull/3219
* Workflow Instance UI Display Fix by @Dioltas95 in https://github.com/elsa-workflows/elsa-core/pull/3220
* Add extensions for eager join option by @martinhjulstrom in https://github.com/elsa-workflows/elsa-core/pull/3233
* Prepend HttpEndpoint's path with BasePath in swagger by @akunzai in https://github.com/elsa-workflows/elsa-core/pull/3236
* Rename file to match the `RedisCacheSignal` class by @akunzai in https://github.com/elsa-workflows/elsa-core/pull/3241
* PR: SignalR Hub Connection Events by @Nico-VanHaaster in https://github.com/elsa-workflows/elsa-core/pull/3242
* #2857 attempt to reapply fix for dropdown by @mohdali in https://github.com/elsa-workflows/elsa-core/pull/3216
* Bump moment from 2.29.2 to 2.29.4 in /src/designer/elsa-workflows-studio by @dependabot in https://github.com/elsa-workflows/elsa-core/pull/3243
* HTTP Endpoint activity Xml support by @tomy2105 in https://github.com/elsa-workflows/elsa-core/pull/3246
* Cleaning up and adding simple support for XML response in HTTP Send Request Activity by @tomy2105 in https://github.com/elsa-workflows/elsa-core/pull/3245
* Credential manager by @andrewkorol in https://github.com/elsa-workflows/elsa-core/pull/2897
* revert port number by @mohdali in https://github.com/elsa-workflows/elsa-core/pull/3259
* Added Tenant Signaler Implementation and Options by @jruckert in https://github.com/elsa-workflows/elsa-core/pull/3250
* Fixing faulted activity display in Journal as well as workflow that remain in running state after fault. by @tomy2105 in https://github.com/elsa-workflows/elsa-core/pull/3251
* Prevent rescheduling of looping/scoped activities if faulted by @tomy2105 in https://github.com/elsa-workflows/elsa-core/pull/3252
* Ability to hide output properties with other misc fixes by @tomy2105 in https://github.com/elsa-workflows/elsa-core/pull/3249
* #3257 Move to CreateAsyncScope by @Sverre-W in https://github.com/elsa-workflows/elsa-core/pull/3271
* RPA Activities Enhancements by @lucapisano in https://github.com/elsa-workflows/elsa-core/pull/3301
* Fix StateMachine by @tchivs in https://github.com/elsa-workflows/elsa-core/pull/3313
* fix error when TargetType not configured #3308 by @jxnkwlp in https://github.com/elsa-workflows/elsa-core/pull/3309
* Retention module improvement v2 #3253 by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/3286
* Update `WorkflowDefinition` model to mapping `CreatedAt` property from API by @akunzai in https://github.com/elsa-workflows/elsa-core/pull/3279
* Feature/extend elsa client by @DSPLX in https://github.com/elsa-workflows/elsa-core/pull/3316

## New Contributors
* @VahidNaderi made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3117
* @AdamJohnSwan made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3121
* @r-ulak made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3139
* @stuartmcgillivray made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3116
* @johnwc made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3164
* @holidayfun made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3181
* @Juandavi1 made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3190
* @smartwaystudio made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3182
* @SebastianWachsmuth made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3212
* @Dioltas95 made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3219
* @Nico-VanHaaster made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3242
* @Sverre-W made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3271
* @tchivs made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3313
* @jxnkwlp made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3309
* @DSPLX made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/3316

**Full Changelog**: https://github.com/elsa-workflows/elsa-core/compare/2.8.2...2.9.0

<hr /><em>This discussion was created from the release <a href='https://github.com/elsa-workflows/elsa-core/releases/tag/2.9.0'>Elsa 2.9.0</a>.</em>

**Comment by johnwc** [comment]
Nice to see the Credential Manager added! How do we start using it, where is the documentation for it? How do we use it in custom activities? Does it support MySQL as a persistent store?

**Reply by sfmskywalker** [reply]
@constantine-v @jwillmer Any chance we could get a small readme? 😃

**Reply by johnwc** [reply]
@constantine-v @jwillmer Any update?
